emergence_lib/src/cursor.rs

```diff
@@ -53,9 +53,18 @@ impl Default for CursorWorldPos {
 }
 
 /// The tile position of the mouse cursor, if it lies over the map.
-#[derive(Resource, Default, Clone, Copy, Deref, DerefMut)]
+#[derive(Resource, Default, Clone, Copy)]
 pub struct CursorTilePos(Option<TilePos>);
 
+impl CursorTilePos {
+    /// The position of the cursor in hex coordinates, if it is on the hex map.
+    ///
+    /// If the cursor is outside the map, this will return `None`.
+    pub fn maybe_tile_pos(&self) -> &Option<TilePos> {
```

Comment:
Feels weird that this returns an option? Seems annoying to work with, and not actually faster for a small type.

Reply:
What should it return instead? It's not guaranteed to exist when the cursor is off the map.
Or do you mean that it returns a reference?

Reply:
I thought the reference was necessary, I didn't realize that `TilePos` was `Copy`.
Nice catch!